client/src/partials/group_invoice/group_invoice.js

```diff
@@ -1,193 +1,217 @@
-        return connect.fetch('/journal/group_invoice/' + id);
-      })
-      .then(function () {
-		    messenger.success($translate.instant('GROUP_INVOICE.SUCCES'));
-      });
-    };
-
-    function formatItems (id) {
-      var items = [];
-      $scope.paying.forEach(function (i) {
-        var item = {};
-        item.uuid = uuid();
-        item.cost = i.payment;
-        item.invoice_uuid = i.inv_po_id;
-        item.payment_uuid = id;
-        items.push(item);
+    // turn on loading
+    $scope.loading = true;
+
+    validate.process(dependencies)
+    .then(setUpModels)
+    .finally(function () {
+      $scope.loading = false;
+    });
+
+    $scope.hasDebitor = true;
+    $scope.action = 'info';
+  };
+
+  // bind data to modules
+  function setUpModels(models) {
+    angular.extend($scope, models);
+
+    $scope.currency = models.currency.data[0];
+
+    if ($scope.invoices) {
+
+      console.log('$scope.invoices', $scope.invoices, $scope.selected.debitor);
+
+      // FIXME: this is hack
+      $scope.invoices.data = $scope.invoices.data.filter(function (d) {
+        return d.balance !== 0;
       });
 
-      return items;
+      // proper formatting
+      $scope.invoices.data.forEach(function (i) {
+        i.invoiceRef = i.abbr + ' ' + i.reference;
+      });
     }
 
-    $scope.filter = function (invoice) {
-      return invoice.balance > 0;
-    };
+    $scope.payment = {};
   }
-]);
+
+  $scope.examineInvoice = function (invoice) {
+    $scope.examine = invoice;
+    $scope.old_action = $scope.action;
+    $scope.action = 'examine';
+  };
+
+  $scope.back = function () {
+    $scope.action = $scope.old_action;
+  };
+
+  $scope.selectConvention = function () {
+    $scope.action = 'selectConvention';
+    $scope.original_id = $scope.data.invoice.convention_id;
+  };
+
+  $scope.saveConvention = function () {
+    $scope.action = '';
+  };
+
+  $scope.resetConvention = function () {
+    $scope.data.invoice.convention_id = $scope.original_id;
+    $scope.action = 'default';
+  };
+
+  $scope.enqueue = function (idx) {
+    var invoice = $scope.invoices.data.splice(idx, 1)[0];
+    invoice.payment = invoice.balance; // initialize payment to be the exact amount -- 100%
+    $scope.paying.push(invoice);
+    $scope.action = 'pay';
+  };
+
+  $scope.dequeue = function () {
+    var total_payment = $scope.total_payment = 0;
+    $scope.paying.forEach(function (i) {
+      $scope.invoices.data.push(i);
+      total_payment += i.payment;
+    });
+    $scope.paying.length = 0;
+    $scope.action = '';
+  };
+
+  $scope.pay = function () {
+    var payment = $scope.payment;
+    payment.project_id = $scope.project.id;
+    payment.group_uuid = $scope.selected.convention.uuid;
+    payment.debitor_uuid  = $scope.selected.debitor.uuid;
+    payment.total = $scope.paymentBalance;
+    payment.date = new Date().toISOString().slice(0,10);
+    $scope.action = 'confirm';
+  };
+
+  $scope.$watch('paying', function () {
```

Comment:
Actually, you don't even need to use `$scope` for this as well.   This was a poor design, and never should have used a `$watch`.

I wasn't going to do the full code rewrite, since the issue (#767) only concerned itself with the UI.  Would you prefer that I rewrite this module?